pkg/controller/clusterdeployment/clusterdeployment_controller.go

```diff
@@ -165,6 +165,10 @@ func (r *ReconcileClusterDeployment) Reconcile(request reconcile.Request) (recon
 	}
 
 	if cd.DeletionTimestamp != nil {
+		if cd.Spec.PreserveOnDelete {
+			cdLog.Warn("skipping deprovisioning due to PreserveOnDelete")
+			r.Delete(context.TODO(), cd)
```

Comment:
We don't want to delete the cluster deployment, if it has a deletion timestamp it's already been deleted. Kubernetes will clean it up when it can. I don't think we need this block rather we can likely handle it in syncDeletedClusterDeployment.

Reply:
@dgoodwin : if I don't delete the unit test fails,  it gets the cd (i.e Assert.Nil(cd) causes it to fail, i think kubernetes hasn't cleaned it up yet)